Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1551,6 +1552,27 @@ def __eq__(self, other):
         except CustomError:
             pass
 
+    def test_weak_cache_descriptor_use_after_free(self):
+        class BombDescriptor:
+            def __get__(self, obj, owner):
+                return {}
+
+        class EvilZoneInfo(self.klass):
+            pass
+
+        EvilZoneInfo._weak_cache = BombDescriptor()
+
+        zone1 = EvilZoneInfo("America/Los_Angeles")
+
+        self.assertIsNotNone(zone1)
+        self.assertEqual(str(zone1), "America/Los_Angeles")
```

Comment:
```suggestion
        self.assertEqual(str(zone1), "America/Los_Angeles")
```
`assertIsNotNone` is superseded by the second assert.